### PR TITLE
Modify KdTree for FLANN backend and UIN32, INT64 indices

### PR DESCRIPTION
**Pull request by kunaltyagi**
Release candidate blocked due to compilation error (no overload exists in flann for non-int/size_t index)

### CODE REVIEW

**Comment by kunaltyagi** [comment]
Turned out to be trickier than I thought

**Comment by mvieth** [comment]
The added `PCL_WARN` seems to be a big problem since it is printed millions of times. Maybe it is possible to just print it once when creating the `KdTreeFLANN` object?
In case you haven't seen it/don't know about it: only `size_t` is directly supported, `int` is only supported via a copy (see e.g. [here](https://github.com/flann-lib/flann/blob/master/src/cpp/flann/algorithms/nn_index.h#L368-L384)). So we actually only have to deal with `size_t` for non-copy, and everything else for copy.

**Comment by kunaltyagi** [comment]
I didn't notice the int. And you're right, I'll move the PCL_WARN into the ctor

**Comment by kunaltyagi** [comment]
Some weird output on the Release CI. Checking that out

**Comment by kunaltyagi** [comment]
Release CI runs only on master. That's why. Can't test on CI unless merged. Works on my laptop though

**Comment by mvieth** [comment]
There seems to be a similar problem in https://github.com/PointCloudLibrary/pcl/blob/master/search/include/pcl/search/impl/flann_search.hpp
What do you think about switching the indices CI from 64 bit unsigned to 32 bit unsigned or 64 bit signed to check if everything is solved, then switch back just before merging?

**Comment by kunaltyagi** [comment]
@larshg @mvieth PTAL

**Comment by larshg** [comment]
LGTM

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp
Comment (mvieth) [inline kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp:44]: What is this for?

**Review by mvieth — COMMENTED** [review]
Just two comments, and then it can be merged after undoing the CI change

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp
Comment (kunaltyagi) [reply kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp:60]: Makes sense

File: kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp
Comment (kunaltyagi) [reply kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp:44]: Nothing anymore

**Comment by kunaltyagi** [comment]
Will revert the CI change once the CI goes green again

**Comment by kunaltyagi** [comment]
Reverting CI

**Comment by kunaltyagi** [comment]
CI was green, reorganized commits:
* KdTree only
* KdTree + Search
* Message
* Support for vector-vector

**Review by mvieth — APPROVED** [review]

**Pull request merged**
